src/__tests__/browserSuites/flag-sets.spec.js

```diff
@@ -0,0 +1,204 @@
+    const splitFilters = [{ type: 'bySet', values: ['set_1','set_2'] }];
+    factory = SplitFactory({ ...baseConfig, sync: { splitFilters }});
+    await factory.client().ready();
+    manager = factory.manager();
+
+  }, 'Polling - SDK with sets configured updates flags according to sets');
+
+  t.test(async (assert) => {
+    let factory;
+    let manager;
+
+    // Receive split change with 1 split belonging to set_1 & set_2 and one belonging to set_3
+    fetchMock.getOnce(baseUrls.sdk + '/splitChanges?since=-1',  function () {
+      return { status: 200, body: splitChange2};
+    });
+
+    // Receive split change with 1 split belonging to set_1 only
+    fetchMock.getOnce(baseUrls.sdk + '/splitChanges?since=1602796638344',  function () {
+      // stored feature flags before update
+      const storedFlags = manager.splits();
+      assert.true(storedFlags.length === 2, 'every feature flag should be added');
+      assert.true(storedFlags[0].name === 'workm');
+      assert.true(storedFlags[1].name === 'workm_set_3');
+      assert.deepEqual(storedFlags[0].sets, ['set_1','set_2']);
+      assert.deepEqual(storedFlags[1].sets, ['set_3']);
+
+      // send split change
+      return { status: 200, body: splitChange1};
+    });
+
+    // Receive split change with 1 split belonging to set_3 only
+    fetchMock.getOnce(baseUrls.sdk + '/splitChanges?since=1602797638344',  function () {
+      // stored feature flags before update
+      const storedFlags = manager.splits();
+      assert.true(storedFlags.length === 2);
+      assert.true(storedFlags[0].name === 'workm');
+      assert.true(storedFlags[1].name === 'workm_set_3');
+      assert.deepEqual(storedFlags[0].sets, ['set_1'], 'the feature flag should be updated');
+      assert.deepEqual(storedFlags[1].sets, ['set_3'], 'the feature flag should remain as it was');
+
+      // send split change
+      return { status: 200, body: splitChange0};
+    });
+
+    fetchMock.getOnce(baseUrls.sdk + '/splitChanges?since=1602798638344',  async function () {
+      // stored feature flags before update
+      const storedFlags = manager.splits();
+      assert.true(storedFlags.length === 2);
+      assert.true(storedFlags[0].name === 'workm');
+      assert.true(storedFlags[1].name === 'workm_set_3');
+      assert.deepEqual(storedFlags[0].sets, ['set_3'], 'the feature flag should be updated');
+      assert.deepEqual(storedFlags[1].sets, ['set_3'], 'the feature flag should remain as it was');
+      await factory.client().destroy();
+      assert.end();
+      return { status: 200, body: {} };
+    });
+
+    // Initialize a factory with polling and no sets configured.
+    factory = SplitFactory(baseConfig);
+    await factory.client().ready();
+    manager = factory.manager();
+
+  }, 'Poling - SDK with no sets configured does not take sets into account when updating flags');
+
+  // EVALUATION
+
+  t.test(async (assert) => {
+    fetchMock.reset();
+    assert.plan(8);
```

Comment:
You can remove `assert.plan(8);`, because all asserts are executed sequentially, so there is no need to count them.
Remove others `assert.plan` too.